api/authRouter.js

```diff
@@ -0,0 +1,105 @@
+const express = require("express");
+const authRouter = express.Router();
+const jwt = require("jsonwebtoken");
+const passport = require("passport");
+const keys = require("../config/keys");
+const User = require("../models/userModel");
+
+/* POST login. */
+authRouter.post("/login", function(req, res, next) {
+  passport.authenticate("login", { session: false }, (err, user, info) => {
+    if (err || !user) {
+      return res.status(400).json({
+        message: info.message,
+        user: user
+      });
+    }
+    req.login(user, { session: false }, err => {
+      if (err) {
+        res.send(err);
+      }
+      // generate a signed son web token with the contents of user object and return it in the response
+      let token = jwt.sign({ id: user._id }, keys.session.cookieKey, {
+        expiresIn: "20h"
+      });
+      return res.json({ token });
+    });
+  })(req, res);
+});
+
+//post signup
+authRouter.post("/signup", (req, res, next) => {
+  if (req.body.password === req.body.passwordVal) {
+    if (req.body.username.length >= 4) {
```

Comment:
Instead of wrapping all the logic with these checks, I would have them as conditionals at the top of the block.  For example:
```
if (req.body.username.length < 4) {
     res.status(400).json({
        success: false,
        message: "username must be at least 4 characters long."
      });
     return;
}
```